.github/workflows/ci-build.yaml

```diff
@@ -14,7 +14,7 @@ concurrency:
   cancel-in-progress: true
 
 permissions:
-  contents: read
+  contents: write
```

Comment:
Per #12035 and #9552, the top-level permissions of a GHA Workflow should _never_ be changed. It should always be read-only.
Only individual jobs should have their permissions changed.
This change will fail OpenSSF / CLOMonitor checks as is.

Also all permissions should have explicit comments about why they're enabled

Reply:
It would also be good to ensure Yarn runs without scripts if we're adding write permissions (similar to [this line](https://github.com/foxglove/studio/blob/0977e3f553d0fc7bed19eabfad477536c32c6b96/.github/workflows/dependabot-fix.yml#L25), although the env var might be different from Yarn v1)

Reply:
I thought the bot would need this permission to commit, but I was mistaken.